core.cc

```diff
@@ -44,18 +44,26 @@ static char ident[1024];
 // wrap: void openlog(const char *ident, int option, int facility);
 NAN_METHOD(OpenLog) {
 
-    // openlog requires ident be statically allocated. Write doesn't guarantee
-    // NULL-termination, so preserve last byte as NULL.
-    info[0]->ToString()->WriteUtf8(ident, sizeof(ident)-1);
-    int option = info[1]->Int32Value();
-    int facility = info[2]->Int32Value();
+    const Nan::Utf8String arg0(info[0]);
+    Local<Int32> arg1 = Nan::To<Int32>(info[1]).ToLocalChecked();
+    Local<Int32> arg2 = Nan::To<Int32>(info[2]).ToLocalChecked();
+
+    // openlog() requires ident be statically allocated.
+    size_t length = arg0.length();
+    if(length) {
+      if(length > sizeof(ident)-1)
+        length = sizeof(ident)-1;
+      strncpy(ident, *arg0, length);
+    }
+    int option = arg1->Value();
+    int facility = arg2->Value();
 
     openlog(ident, option, facility);
 
     return;
 }
 
-static char* dupBuf(const Handle<Value>& arg) {
+static char* dupBuf(const Local<Value>& arg) {
```

Comment:
@bnoordhuis this surprised me, perhaps Local used to derive from Handle, and now it doesn't?

Reply:
That's right.